src/MusicStore/Program.cs

```diff
@@ -17,14 +17,9 @@ public Task<int> Main(string[] args)
             var config = new Configuration();
             config.AddCommandLine(args);
 
-            var context = new HostingContext()
-            {
-                Configuration = config,
-                ServerFactoryLocation = "Microsoft.AspNet.Server.WebListener",
-                ApplicationName = "MusicStore"
-            };
-
-            using (new HostingEngine().Start(context))
+            using (WebApplication.CreateHostingEngine(config)
```

Comment:
T is the startup type?  @lodejard suggested that belongs as a TestServer extension instead, I can bring it up with him again